client/webserver/site/src/js/doc.ts

```diff
@@ -110,6 +110,7 @@ export default class Doc {
    * the bounds of the specified element.
    */
   static mouseInElement (e: MouseEvent, el: HTMLElement): boolean {
+    if (e.target && (e.target as HTMLElement).tagName === 'OPTION') return true
     const rect = el.getBoundingClientRect()
     return e.pageX >= rect.left && e.pageX <= rect.right &&
       e.pageY >= rect.top && e.pageY <= rect.bottom
```

Comment:
Seems like we're overthinking this.

```js
  static mouseInElement (e: MouseEvent, el: HTMLElement): boolean {
    return el.contains(e.target as Node)
  }
```

Reply:
Nice find. That should work, with a couple of caveats.

1. You probably need to check if the target is the element itself, otherwise that click padding would become ineffective. Something like
```Javascript
return el === e.target || el.contains(e.target as Node)
```
2. This would fail if we had an element that's not a descendant absolutely positioned over the form. I'm not sure that we do that with anything but tooltips, so it probably won't matter to us.

Reply:
> 1. You probably need to check if the target is the element itself, otherwise that click padding would become ineffective. Something like

https://developer.mozilla.org/en-US/docs/Web/API/Node/contains

contains covers self apparently:

> The contains() method of the [Node](https://developer.mozilla.org/en-US/docs/Web/API/Node) interface returns a boolean value indicating whether a node is a descendant of a given node, that is the node itself, one of its direct children ([childNodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/childNodes)), one of the children's direct children, and so on.
> 
> Note: A node is contained inside itself.

Reply:
> contains covers self, apparently

yeah, your suggestion is efficient.